src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = /[.\-/]/;
+  const newDate = [];
+  const objectDate = {};
+  const splitDate = date.split(oldSeparator);
+  const isPunctuation = (str) => /^[\p{P}\p{S}]+$/u.test(str);
+  const punct = toFormat.filter(isPunctuation).join('');
```

Comment:
The `punct` variable is derived from filtering `toFormat` with `isPunctuation`, which is unnecessary complexity. Instead, directly assign the separator to a variable like `const newSeparator = toFormat[toFormat.length - 1]`.